src/apiConfigProvider.component.test.tsx

```diff
@@ -0,0 +1,108 @@
+  return <div data-testid="settings">{JSON.stringify(settings)}</div>;
+};
+
+describe('APIConfigProvider', () => {
+  beforeEach(() => {
+    global.document.dispatchEvent = vi.fn();
+    global.CustomEvent = vi.fn();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  // Create a wrapper for our settings tests.
+  const renderComponent = (): RenderResult =>
+    renderComponentWithRouterProvider(
+      <APIConfigProvider>
+        <APIConfigTest />
+      </APIConfigProvider>
+    );
+
+  it('settings are loaded (with spares)', async () => {
+    renderComponent();
+
+    // Preloader is in a loading state when ConfigProvider is
+    // loading the configuration.
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+    });
+    expect(screen.getByTestId('settings')).toBeInTheDocument();
+    expect(screen.getByTestId('settings')).toHaveTextContent(
+      JSON.stringify({
+        spares: {
+          sparesFilterState:
+            '?state=N4IgxgYiBcDaoEsAmMQGcCeaAuBTAtgHTYYAOuhAbgIYA2ArriADQg0NNygnmo4BOCAHYBzFmzqNUAZWwB7ftRFMAvgF11KoA',
+          sparesColumnsFilters: {
+            cF: [
+              {
+                id: 'system.type.value',
+                value: [{ type: 'string', value: 'Storage' }],
+              },
+            ],
+          },
+          isLoading: false,
+          sparesDefinition: { system_types: [{ id: '1', value: 'Storage' }] },
+        },
+      })
+    );
+  });
+
+  it('settings are loaded (without spares)', async () => {
+    server.use(
+      http.get('/v1/settings/spares-definition', () => {
+        return HttpResponse.json({ system_types: [] }, { status: 200 });
+      })
+    );
+    renderComponent();
+
+    // Preloader is in a loading state when ConfigProvider is
+    // loading the configuration.
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+    });
+    expect(screen.getByTestId('settings')).toBeInTheDocument();
+    expect(screen.getByTestId('settings')).toHaveTextContent(
+      JSON.stringify({})
+    );
+  });
+
+  it('settings are loaded (without spares 204 no content)', async () => {
+    server.use(
+      http.get('/v1/settings/spares-definition', () => {
+        return HttpResponse.json(undefined, { status: 204 });
+      })
+    );
+    renderComponent();
+
+    // Preloader is in a loading state when ConfigProvider is
+    // loading the configuration.
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+    });
+    expect(screen.getByTestId('settings')).toBeInTheDocument();
+    expect(screen.getByTestId('settings')).toHaveTextContent(
+      JSON.stringify({})
+    );
+  });
```

Comment:
It cant return an empty list, only the 204, or a list of values, so I would just do the one test personally.